lib/datadog/grok/src/parse_grok.rs

```diff
@@ -85,7 +85,12 @@ fn apply_grok_rule(source: &str, grok_rule: &GrokRule, remove_empty: bool) -> Re
                             .expect("field does not exist")
                             .cloned()
                         {
-                            Some(Value::Array(mut values)) => values.push(value),
+                            Some(Value::Array(mut values)) => {
+                                values.push(value);
+                                parsed.target_insert(field, values.into()).unwrap_or_else(
+                                    |error| warn!(message = "Error updating field value", field = %field, %error)
+                                );
+                            }
```

Comment:
Just a note that once `target_get_mut` lands in https://github.com/vectordotdev/vector/pull/12576, this could be made more performant by doing the mutation directly, instead of getting the value, cloning it, and then re-inserting it.